src/main/java/me/dinowernli/grpc/polyglot/command/ServiceList.java

```diff
@@ -25,22 +25,42 @@ public static void listServices(
       String protoDiscoveryRoot,
       Optional<String> serviceFilter,
       Optional<String> methodFilter,
-      Optional<Boolean> withMessage) {
+      Optional<Boolean> withMessage,
+      Optional<String> listOutputFormat) {
 
     ServiceResolver serviceResolver = ServiceResolver.fromFileDescriptorSet(fileDescriptorSet);
 
-    // Add white-space before the rendered output
-    output.newLine();
+    if (listOutputFormat.isPresent() && listOutputFormat.get().equals("json"))  output.write("[\n"); else output.newLine();
+
+    // Dealing with commas for JSON
+    boolean firstServiceHasBeenPrinted = false;
 
     for (ServiceDescriptor descriptor : serviceResolver.listServices()) {
       boolean matchingDescriptor =
           !serviceFilter.isPresent()
           || descriptor.getFullName().toLowerCase().contains(serviceFilter.get().toLowerCase());
 
       if (matchingDescriptor) {
```

Comment:
Almost there :) Any chance we can just print the json of descriptor.toProto()? That should give you the information you need and avoid manually constructing json output.

Reply:
I could try but will need some help. As I think I mentioned before as far as I can tell using the JsonPrinters do not do what we need. eg.
`
printing:String jsonFormat = JsonFormat.printer().print(descriptor.toProto().toBuilder());
`
Returns something similar to:
`{
"name": "TestMethod",
"inputType": ".polyglot.test.TestRequest",
"outputType": ".polyglot.test.TestResponse",
"options": {}
}
`
There is not enough information about the types. I imagine you could recursively resolve every type which is more than one level deep to get down to the primitive level of strings, int etc. but this will require a decent change to the code. 
I think this approach is the simplest method given what's already written but if you know a better way then let me know.